iceoryx_dust/include/iceoryx_dust/posix_wrapper/named_pipe.hpp

```diff
@@ -150,9 +151,7 @@ class NamedPipe : public DesignPattern::Creation<NamedPipe, IpcChannelError>
         static constexpr units::Duration WAIT_FOR_INIT_SLEEP_TIME = units::Duration::fromMilliseconds(1);
 
         std::atomic<uint64_t> initializationGuard{INVALID_DATA};
-        /// NOLINTJUSTIFICATION @todo iox-#1614 replace with cxx::array implementation
-        /// NOLINTNEXTLINE(hicpp-avoid-c-arrays,cppcoreguidelines-avoid-c-arrays)
-        cxx::optional<UnnamedSemaphore> semaphores[2U];
+        containers::UninitializedArray<cxx::optional<UnnamedSemaphore>, 2> semaphores;
```

Comment:
I think option 2 is hard to achieve because of the semaphore builder but please correct if I overlook something.
Regarding option 1: Can we have also two separate semaphores (inside an optional)? If not can you help me with clang-tidy justification?

Reply:
Ouh you are right, this is less then optional ;)

Reply:
About the clang-tidy justification, what about the todo with the `cxx::array` implementation - would this suffice for now since it is in dust? 
Otherwise we could also create two member variables and call them `sendSemaphore` and `receiveSemaphore`.

What do you think?

Reply:
I created two member variables.